app/python_apps/aiohttp_ratelimit.py

```diff
@@ -0,0 +1,103 @@
+# A demo case of async request get with rate limit.
+# pip install aiohttp aiohttp-client-manager bucketratelimiter tqdm
+from typing import Tuple, Any, Union, List, Coroutine
+
+import asyncio
+from asyncio import events, coroutines
+from datetime import datetime
+from asyncio import tasks
+
+from bucketratelimiter import AsyncioBucketTimeRateLimiter
+import aiohttp_client
+from tqdm.asyncio import tqdm
+
+
+async def async_tasks(cors: List[Coroutine]) -> List:
+    results = []
+    with tqdm(total=len(cors)) as pbar:
+        for next_to_complete in asyncio.as_completed(cors):
+            answer = await next_to_complete
+            results.append(answer)
+            pbar.update()
+    return results
+
+
+def cancel_all_async_tasks(loop):
+    to_cancel = tasks.all_tasks(loop)
+    if not to_cancel:
+        return
+
+    for task in to_cancel:
+        task.cancel()
+
+    loop.run_until_complete(tasks.gather(*to_cancel, loop=loop, return_exceptions=True))
+
+    for task in to_cancel:
+        if task.cancelled():
+            continue
+        if task.exception() is not None:
+            loop.call_exception_handler(
+                {
+                    "message": "unhandled exception during asyncio.run() shutdown",
+                    "exception": task.exception(),
+                    "task": task,
+                }
+            )
+
+
+class AsyncApi:
+    limiter = AsyncioBucketTimeRateLimiter(max_size=10, recovery_time=5.0, rest_time=0.5)
```

Comment:
@you-n-g, do you have any suggestions to support run-time rate-limit assignment?
That will be better if we can support Something like AsnycApi(max_size=10, recovery_time=5.0, rest_time=0.5) in the constructor.

Reply:
I think the issue that blocks you is making the decorator work rightly for get_data.

I think `@property` can solve this.  It can return a decorated function and then be called.

Reply:
That will be great if you can write a patch for it. I can learn from you.
I would like to read this(https://book.pythontips.com/en/latest/decorators.html) when I have more free time. :)

Reply:
I implemented the patch just now
Please check if it works for you.
https://github.com/you-n-g/cheatsheets/pull/4